validator/db/filesystem/db.go

```diff
@@ -0,0 +1,399 @@
+package filesystem
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/io/file"
+	validatorpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1/validator-client"
+	"gopkg.in/yaml.v3"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	backupsDirectoryName = "backups"
+
+	slashingProtectionDirName = "slashing-protection"
+	configurationFileName     = "configuration.yaml"
+
+	DatabaseDirName = "validatorClientData"
+)
+
+type (
+	// Store is a filesystem implementation of the validator client database.
+	Store struct {
+		configurationMu    sync.RWMutex
+		pkToSlashingMu     map[[fieldparams.BLSPubkeyLength]byte]*sync.RWMutex
```

Comment:
Mutexes in go are designed to be used without pointers, as using pointers can lead to issues. If this gets copied, for example, you will have two pointers to the same mutex. The default value of a mutex is also valid, so it should be safe to keep it as such

Reply:
Maybe I missed something, but I need to mutate mutexes in the map (`mu.RLock()` and `mu.RUnlock()`).

https://github.com/uber-go/guide/blob/master/style.md#receivers-and-interfaces